elrond_dex_farm/src/lib.rs

```diff
@@ -118,8 +152,20 @@ pub trait Farm:
             / farm_attributes.total_liquidity_amount.clone();
         require!(enter_amount > 0, "Cannot exit farm with 0 entering amount");
 
+        // For locked rewards option, exit before minumum epochs is not allowed
+        require!(
+            !farm_attributes.with_locked_rewards
+                || !self.should_apply_penalty(farm_attributes.entering_epoch),
+            "Exit too early for lock rewards option"
+        );
+        require!(
+            !farm_attributes.with_locked_rewards
+                || enter_amount >= Self::BigUint::from(LOCKED_REWARDS_LIQUIDITY_MUTIPLIER),
```

Comment:
why are you comparing with the value "2" instead of value 0 ?

Reply:
because of the division happening a few line below. if the initial value is 1, then after the division with 2 it will still be 0

Reply:
refactored to be more intuitive